src/types.jl

```diff
@@ -241,73 +536,101 @@ eltype(::Type{StrSxp}) = Ptr{CharSxp}
-released via `unprotect`. Returns the same pointer, allowing inline use.
+    protect(p::Ptr{<:Sxp})
+
+Stack-based protection of garbage collection of R objects.
 
-This is faster than `preserve`, but more restrictive. Really only useful
-inside functions.
+Objects are released via [`unprotect`](@ref).
+Returns the same pointer, allowing inline use.
+
+This is faster than [`preserve`](@ref), but more restrictive.
+Really only useful inside functions, where you can control the `unprotect` step.
 """
 protect(p::Ptr{S}) where S<:Sxp = ccall((:Rf_protect,libR), Ptr{S}, (Ptr{S},), p)
 
 """
-Release last `n` objects gc-protected by `protect`.
+    unprotect(n)
+
+Release last `n` objects GC-protected by [`protect`](@ref).
 """
 unprotect(n::Integer) = ccall((:Rf_unprotect,libR), Nothing, (Cint,), n)
 
 """
-The SEXPTYPE number of a `Sxp`
+    sexpnum(s::Sxp)
+    sexpnum(p::Ptr{<:Sxp})
+
+Return the `SEXPTYPE` number, i.e. type tag, of a `Sxp`.
 
 Determined from the trailing 5 bits of the first 32-bit word. Is
-a 0-based index into the `info` field of a `SxpHead`.
+a 0-based index into the `info` field of a [`SxpHead`](@ref).
 """
 sexpnum(h::SxpHead) = h.info & 0x1f
 sexpnum(p::Ptr{S}) where S<:Sxp = sexpnum(unsafe_load(p))
 
-"vector of R Sxp types"
-const typs = [NilSxp,SymSxp,ListSxp,ClosSxp,EnvSxp,
-              PromSxp,LangSxp,SpecialSxp,BuiltinSxp,CharSxp,
-              LglSxp,Nothing,Nothing,IntSxp,RealSxp,
-              CplxSxp,StrSxp,DotSxp,AnySxp,VecSxp,
-              ExprSxp,BcodeSxp,ExtPtrSxp,WeakRefSxp,RawSxp,
-              S4Sxp]
+"""
+    SXP_TYPES
 
-for (i,T) in enumerate(typs)
+Ordered collection of R `SEXP` types, so that the (index - 1)
+matches the type tag.
+"""
+const SXP_TYPES = (NilSxp, SymSxp, ListSxp, ClosSxp, EnvSxp,
+              PromSxp, LangSxp, SpecialSxp, BuiltinSxp, CharSxp,
+              LglSxp, Nothing, Nothing, IntSxp, RealSxp,
+              CplxSxp, StrSxp, DotSxp, AnySxp, VecSxp,
+              ExprSxp, BcodeSxp, ExtPtrSxp, WeakRefSxp, RawSxp,
+              S4Sxp)
+
+for (i, T) in enumerate(SXP_TYPES)
     if T != Nothing
         @eval sexpnum(::Type{$T}) = $(i-1)
     end
 end
 
 
 """
-Convert a `Ptr{UnknownSxp}` to an appropriate `Ptr{S<:Sxp}`.
+    sexp(p::Ptr{UnknownSxp})
+
+Return a restrictively parameterized `Ptr{<:Sxp}` pointing the same object.
 """
 function sexp(p::Ptr{UnknownSxp})
     typ = sexpnum(p)
     0 ≤ typ ≤ 10 || 13 ≤ typ ≤ 25 || error("Unknown SEXPTYPE $typ")
-    styp = typs[typ+1]
-    Ptr{styp}(p)
+    styp = SXP_TYPES[typ+1]
+    return Ptr{styp}(p)
 end
 sexp(s::Ptr{S}) where S<:Sxp = s
 sexp(r::RObject) = r.p
 
-sexp(::Type{S},s::Ptr{S}) where S<:Sxp = s
-sexp(::Type{S},r::RObject{S}) where S<:Sxp = r.p
+"""
+   sexp(::Type{<:Sxp}, s::RObject{<:Sxp})
+
+Return the associated `Sxp` pointer.
+"""
+sexp(::Type{S}, r::RObject{S}) where S<:Sxp = r.p
+# do we need this method?
+# sexp(::Type{S}, s::Ptr{S}) where S<:Sxp = s
```

Comment:
Probably doesn't hurt and possibly avoids some unnecessary indirection but I can't say for certain

Reply:
Yeah, this is why I left here but commented out. I don't think it breaks anything (and all tests pass...) but just in case we get a complaint in the next few weeks...

I'm hoping to make this less relevant in general by slowing move more to dealing with `RObject` and deprecating/removing/hiding the pointer (`Sxp`) interface.

Reply:
> and all tests pass

Worth noting that the package doesn't have 100% coverage, though I haven't looked at whether this definition was covered.

Reply:
```suggestion
# do we need this method?
sexp(::Type{S}, s::Ptr{S}) where S<:Sxp = s
```

Reply:
re-enable method